src/service/mustwatch_service.py

```diff
@@ -0,0 +1,304 @@
+from typing import Tuple, Dict, List
+
+from sqlalchemy.orm.query import Query
+from telebot.types import CallbackQuery
+
+from src.service.register_service import RegisterService
+from src.view.buttons import Button
+
+
+class MustwatchService:
+
+    def __init__(self, alchemist, users_read, user_requests_read, watches_read, mustwatches_read, watches_redact,
+                 user_requests_redact, mustwatches_redact):
+        self.users_read = users_read
+        self.user_requests_read = user_requests_read
+        self.watches_read = watches_read
+        self.mustwatches_read = mustwatches_read
+        self.watches_redact = watches_redact
+        self.user_requests_redact = user_requests_redact
+        self.mustwatches_redact = mustwatches_redact
+        self.session = alchemist.session
+
+    def __clear_watches_id_list_from_watches_for_all_users(self,
+                                                           chosen_user_tuple: Tuple[int],
+                                                           same_group_watches_id_list: List[int]) -> List[int]:
+        same_group_users_amount = len(chosen_user_tuple)
+        watches_id_list = []
+        for watch in same_group_watches_id_list:
+            if same_group_watches_id_list.count(watch) < same_group_users_amount:
+                watches_id_list.append(watch)
```

Comment:
Also looks like this algorythm has O(n^2) complexity, but for now I cannot suggest you way to reduce it :sweat_smile: